ci-scripts/yocto.groovy

```diff
@@ -0,0 +1,189 @@
+        if (url?.trim()) {
+            vagrant("sed 's|%CACHEURL%|${url}|g' /vagrant/site.conf.in > ${yoctoDir}/build/conf/site.conf")
+        } else {
+            echo "No cache setup"
+        }
+    }
+}
+
+void buildImageAndSDK(String yoctoDir, String imageName, boolean dev=true) {
+    // If we have a site.conf, that means we have caching, if we don't that
+    // means we should do a fetchall.
+    if (vagrant("test -f ${yoctoDir}/build/conf/site.conf", true) != 0) {
+        stage("Fetch sources") {
+            vagrant("/vagrant/cookbook/yocto/fetch-sources-for-recipes.sh ${yoctoDir} ${bitbake_image}")
+        }
+    }
+
+    stage("Bitbake ${imageName}") {
+        vagrant("/vagrant/cookbook/yocto/build-images.sh ${yoctoDir} ${imageName}")
+        if (dev) {
+            vagrant("/vagrant/cookbook/yocto/build-images.sh ${yoctoDir} ${imageName}-dev")
+        }
+    }
+
+    stage("Build SDK ${imageName}") {
+        vagrant("/vagrant/cookbook/yocto/build-sdk.sh ${yoctoDir} ${imageName}")
+        if (dev) {
+            vagrant("/vagrant/cookbook/yocto/build-sdk.sh ${yoctoDir} ${imageName}-dev")
+        }
+    }
+}
+
+void archiveCache(String yoctoDir, boolean archive, String archivePath) {
+    if (archive && archivePath?.trim()) {
+        stage("Archive cache") {
+            vagrant("rsync -trpg ${yoctoDir}/build/downloads/ ${archivePath}/downloads/")
+            vagrant("rsync -trpg ${yoctoDir}/build/sstate-cache/ ${archivePath}/sstate-cache")
+        }
+    }
+}
+
+void archiveArtifacts(String yoctoDir, String suffix) {
+    stage("Archive artifacts") {
+        String artifactDir = "artifacts_${suffix}"
+
+        sh "rm -rf ${artifactDir}"
+        sh "mkdir ${artifactDir}"
+
+        // Copy images and SDK to the synced directory
+        vagrant("/vagrant/ci-scripts/copy_to_archive ${yoctoDir}/build /vagrant/${artifactDir}")
+
+        // And save them in Jenkins
+        archiveArtifacts "${artifactDir}/**"
+    }
+}
+
+void buildWithLayer(String variant_name, String bitbake_image, String layer, String layerPath) {
+    // Store the directory we are executed in as our workspace.
+    String yoctoDir = "/home/yoctouser/pelux_yocto"
+    String manifest = "pelux.xml"
+
+    // We can't do "checkout scm" here since pelux-manifests is not the
+    // canonical repo here.
+    sh "git submodule update --init"
+    try {
+        // Initialize vagrant and repo manifest
+        startVagrant()
+        repoInit(manifest)
+
+        replaceLayer(yoctoDir, layer, layerPath)
+
+        // Setup yocto
+        String templateConf="${yoctoDir}/sources/meta-pelux/conf/variant/${variant_name}"
+        setupBitbake(yoctoDir, templateConf)
+        setupCache(yoctoDir, env.YOCTO_CACHE_URL)
+
+        // Build the images
+        try {
+            buildImageAndSDK(yoctoDir, bitbake_image)
+        } finally { // Archive cache even if there were errors.
+            boolean archive = env.ARCHIVE_CACHE == "true"
```

Comment:
In another change you rewrote this as "env.VAR?.trim" shouldn't that be done here?
(And also for other env vars further down in the code)

Reply:
It could have been done here as well, yes. I felt however that being explicit about stuff being booleans adds some certainty.